### PR TITLE
[dagster-azure] Add support for DefaultAzureCredential for AzureBlobComputeLogManager

### PR DESCRIPTION
**Pull request by mpicard**
### Summary & Motivation

As an Azure user, I want to easily authenticate with Azure in multiple common scenarios using the `azure.identity.DefaultAzureIdentity` so that all my environments, locally and in azure are securely authenticated. 

### How I Tested These Changes

I unit tested this and also manually tested it in a locally and an azure environment.

### CODE REVIEW

**Comment by mpicard** [comment]
@jamiedemaria

**Comment by jamiedemaria** [comment]
Build kite results:
`helm/dagster/schema/schema_tests/test_instance.py` has some failing tests - you can run this test locally so see the issue, but it looks like mismatched config entries due to the new azure credentials

`ruff` formatting error - we recently switched from using `isort` and `pylint` to using `ruff`. I recommend rebasing/merging from master to ensure you have a version of the codebase formatted with `ruff` and then running `make ruff` from the top level dagster directory. This will format your code according to the new standards

**Review by jamiedemaria — CHANGES_REQUESTED** [review]
one small request, but otherwise this looks good!

File: python_modules/libraries/dagster-azure/dagster_azure/blob/compute_log_manager.py [hunk 1]
```diff
@@ -62,18 +69,27 @@ def __init__(
...
     ):
         self._storage_account = check.str_param(storage_account, "storage_account")
         self._container = check.str_param(container, "container")
         self._blob_prefix = self._clean_prefix(check.str_param(prefix, "prefix"))
-        check.str_param(secret_key, "secret_key")
+        self._default_azure_credential = check.opt_dict_param(
+            default_azure_credential, "default_azure_credential"
+        )
+        check.opt_str_param(secret_key, "secret_key")
+
```
[the remark this answers is not in the document]
Comment (mpicard) [reply python_modules/libraries/dagster-azure/dagster_azure/blob/compute_log_manager.py:85]: Sure thing, I made the changes

**Comment by mpicard** [comment]
I updated the helm schema test, it still fails locally, maybe you can help me understand why it's still failing? I seem to have refactored the test properly I believe

**Comment by prha** [comment]
Had a hard time pushing updates to your PR, but here's a patch file that I think should resolve the failing tests:
[azure_creds.patch](https://github.com/dagster-io/dagster/files/10460390/azure_creds.patch)

**Review by jamiedemaria — APPROVED** [review]

**Pull request merged**